### PR TITLE
Implement 'OptionalAssert.containsSame'

### PR DESCRIPTION
**Pull request by nicolaiparlog**
:exclamation: This pull request is a continuation of #430.

And I wondered why GitHub predicted merge conflicts... :blush: The branch should be up to date now but I screwed up while trying to rebase and deleting my branch was the fastest solution. Hence the new pull request.

@joel-costigliola Regarding your input:
* The message for `OptionalShouldContain.shouldContain` is now as it was before (i.e. without mention of equals).
* I created a new message for `shouldContainSame` (see below).
* I'll check out `ShouldBeEqual` now...
* Didn't initially see `OptionalShouldContain_create_Test`. I added a test.

I am still unsure about the new message, though. In your proposal `expected` is there twice - are you sure that's best?

### CODE REVIEW

File: src/test/java/org/assertj/core/error/OptionalShouldContain_create_Test.java [hunk 1]
```diff
@@ -71,4 +72,14 @@ public void should_create_error_message_with_optionallong() {
                                        "  <10L>\n" +
                                        "but did not.");
   }
+
+  @Test
+  public void should_create_error_message_for_different_instances() {
+    String errorMessage = shouldContainSame(Optional.of(new Integer(10)), new Integer(10)).create();
+    assertThat(errorMessage).isEqualTo("\nExpecting:\n" +
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/error/OptionalShouldContain_create_Test.java]: even though, it is done yet done for the other assertion,we should use `String.format` and `%n` to have proper line separator when running the tests from different OS, it should look like:

```java
   assertThat(message).isEqualTo(format("[Test] %n" +
                                         "Expecting:%n" +
                                         " <[\"Yoda\"]>%n" +
                                         "to contain:%n" +
                                         " <[\"Luke\", \"Yoda\"]>%n" +
                                         "but could not find:%n" +
                                         " <[\"Luke\"]>%n"));
 ```

I won't be mad if you decide to change the other tests assertions messages :smile_cat:

**Comment by joel-costigliola** [comment]
BTW, forget my suggested message, I was thinking to another assertion, yours is fine.

File: src/test/java/org/assertj/core/error/OptionalShouldContain_create_Test.java (hunk 1 above)
Comment (nicolaiparlog) [inline src/test/java/org/assertj/core/error/OptionalShouldContain_create_Test.java]: I wondered about that. As a Windows user I actually ran into failing tests here, but figured this was the AssertJ-style. :wink:

The whole class uses `%n` now.

**Comment by joel-costigliola** [comment]
Sorry it didn't make it for 3.1.0, I forgot about it because I did not set the Java 8 label that I use to track java 8 issues.
It will be in the 3.2.0 release.

**Comment by nicolaiparlog** [comment]
No worries. :smiley: I updated the PR so it can be merged.

**Comment by nicolaiparlog** [comment]
Is there anything I can do to move this pull request along?

**Comment by joel-costigliola** [comment]
not really, I just need to review it. I'm juts busy with a new job and the assertion generator but I will find time by the end of the month.

**Comment by joel-costigliola** [comment]
Integrated, thanks.

**Pull request closed without merging**